pyquil/tests/test_quil.py

```diff
@@ -114,6 +114,13 @@ def test_plus_operator():
     assert p.out() == "H 0\nX 0\nY 0\nZ 0\n"
 
 
+def test_iteration():
+    program = Program(H(0), Y(1), CNOT(0, 1)).measure(0, 0).if_then(0, Program(X(0)), Program())
+    assert program[0] == (0, H(0))
```

Comment:
Nitpick: I'd suggest breaking this out into two unit tests. One for indexing and one for iteration